src/Frontend/src/views/throughputreport/endpoints/DetectedListView.spec.ts

```diff
@@ -0,0 +1,330 @@
+      const filterNameElement = getFilterNameElement();
+      await user.selectOptions(filterNameTypeElement, "Begins with");
+      await user.clear(filterNameElement);
+      await user.type(filterNameElement, "boo");
+
+      const th = screen.getByText(columnTitle);
+      const table = th.parentElement?.parentElement?.parentElement as HTMLTableElement;
+
+      expect(table.rows.length).toBe(1 + 1 /* includes headers */);
+    });
+  });
+
+  describe("sorting by", async () => {
+    test("throughput", async () => {
+      const columnTitle = "Special column name";
+      const dataLength = 5;
+
+      await renderComponent({ source: DataSource.Broker, indicatorOptions: [UserIndicator.NServiceBusEndpoint], columnTitle: columnTitle }, async (driver) => {
+        await driver.setUp(precondition.hasLicensingEndpoints([...[...Array(dataLength).keys()].map((i) => ({ name: `${i}Beta`, is_known_endpoint: false, user_indicator: UserIndicator.PlannedToDecommission, max_daily_throughput: i }))]));
+      });
+
+      const user = userEvent.setup();
+      await user.click(screen.getByRole("button", { name: /Sort by/i }));
+      await user.click(screen.getByRole("link", { name: "throughput" }));
+
+      const th = screen.getByText(columnTitle);
+      const table = th.parentElement?.parentElement?.parentElement as HTMLTableElement;
+
+      let throughput = 0;
+      for (const row of [...table.rows].slice(1)) {
+        expect(row.cells[1].textContent).toBe(`${throughput++}`);
+      }
+
+      await user.click(screen.getByRole("button", { name: /Sort by/i }));
+      await user.click(screen.getByRole("link", { name: "throughput (Descending)" }));
+
+      throughput = dataLength - 1;
+      for (const row of [...table.rows].slice(1)) {
+        expect(row.cells[1].textContent).toBe(`${throughput--}`);
+      }
+    });
+
+    test("name", async () => {
+      const columnTitle = "Special column name";
+      const names = ["basilisk", "octopus", "hamster", "anteater", "porcupine", "gazelle", "seal", "lynx", "crocodile", "mountain goat", "yak", "polar bear", "horse", "gorilla", "zebu", "salamander", "alligator", "vicuna", "goat", "bunny"];
+
+      await renderComponent({ source: DataSource.Broker, indicatorOptions: [UserIndicator.NServiceBusEndpoint], columnTitle: columnTitle }, async (driver) => {
+        await driver.setUp(precondition.hasLicensingEndpoints([...[...names].map((name, idx) => ({ name, is_known_endpoint: false, user_indicator: UserIndicator.PlannedToDecommission, max_daily_throughput: idx }))]));
+      });
+
+      const user = userEvent.setup();
+      await user.click(screen.getByRole("button", { name: /Sort by/i }));
+      await user.click(screen.getByRole("link", { name: "name" }));
+
+      const th = screen.getByText(columnTitle);
+      const table = th.parentElement?.parentElement?.parentElement as HTMLTableElement;
+
+      let orderedNames = names.sort((a, b) => a.localeCompare(b));
```

Comment:
To avoid having logic leak into the test code, prefer asserting against a fixed expected result for the given set of preconditions (arrangement) the test case is validating. This also makes the test code more straightforward for the reader trying to understand the specific requirement the test is checking.

Reply:
I fixed this